src/server_manager/ui_components/outline-server-settings.html

```diff
@@ -133,7 +161,11 @@ <h3>DigitalOcean</h3>
             <h3>[[localize('settings-server-info')]]</h3>
             <paper-input id="serverNameInput" class="server-name" value="{{name}}" label="[[localize('settings-server-name')]]" always-float-label maxlength="100" on-keydown="_handleNameInputKeyDown" on-blur="_handleNameInputBlur"></paper-input>
             <p class="detail">[[localize('settings-server-rename')]]</p>
-            <paper-input readonly value="[[serverPortForNewAccessKeys]]" label="[[localize('settings-access-key-port')]]" hidden$="[[!serverPortForNewAccessKeys]]" always-float-label maxlength="100"></paper-input>
+            <div id="portForNewAccessKeysContainer">
+              <paper-input id="keysPortInput" value="[[serverPortForNewAccessKeys]]" label="[[localize('settings-access-key-port')]]" maxlength="5" error-color="#ffffff" error-message="ERRORMESSSAGE" hidden$="[[!serverPortForNewAccessKeys]]" on-click="_handleKeysPortClick" on-keyup="_handleKeysPortInputKeyDown" always-float-label maxlength="100"></paper-input>
```

Comment:
We can only allow numeric input on the form by using: `auto-validate allowed-pattern="[0-9]{5}" error-message=[[localize('error-keys-port-bad-input')]]` (double check the regex). Then we can just validate that the port is within range in `_handleKeysPortInputKeyDown`.

We can also use a custom validator (`validator="_validatePortInRange"`) to separate the key stroke handling (esc, enter) and the port range validation.

Reply:
I chose not to use a custom validator since the validator and the server response would be fighting for control of the invalid property: from https://www.webcomponents.org/element/@polymer/paper-input/elements/paper-input#property-invalid

> If autoValidate is true, the invalid attribute is managed automatically, which can clobber attempts to manage it manually.

So I sidestepped the issue.

Reply:
That's fine. I still think we'd benefit from only allowing numeric input: `pattern="[0-9]{5}"`.

Reply:
Done